dali/operators/reader/loader/video/frames_decoder_base.cc

```diff
@@ -128,50 +132,72 @@ std::string FramesDecoderBase::GetAllStreamInfo() const {
   return ss.str();
 }
 
-bool FramesDecoderBase::FindVideoStream(bool init_codecs) {
-  if (init_codecs) {
+bool FramesDecoderBase::SelectVideoStream(int stream_id, bool require_available_avcodec) {
+  if (stream_id < 0 || stream_id >= static_cast<int>(av_state_->ctx_->nb_streams)) {
+    return false;
+  }
+  av_state_->stream_id_ = stream_id;
+  av_state_->codec_params_ = av_state_->ctx_->streams[stream_id]->codecpar;
+
+  LOG_LINE << "Selecting stream " << stream_id
+           << " (codec_id=" << av_state_->codec_params_->codec_id
+           << ", codec_type=" << av_state_->codec_params_->codec_type
+           << ", format=" << av_state_->codec_params_->format
+           << ", width=" << av_state_->codec_params_->width
+           << ", height=" << av_state_->codec_params_->height
+           << ", sample_rate=" << av_state_->codec_params_->sample_rate
+           << ", bit_rate=" << av_state_->codec_params_->bit_rate << ")" << std::endl;
+
+  if (require_available_avcodec) {
+    if ((av_state_->codec_ = avcodec_find_decoder(av_state_->codec_params_->codec_id)) == nullptr) {
+      LOG_LINE << "No decoder found for codec " << avcodec_get_name(av_state_->codec_params_->codec_id)
+               << " (codec_id=" << av_state_->codec_params_->codec_id << ")" << std::endl;
+      return false;
+    }
+  }
+
+  if (av_state_->codec_params_->codec_type != AVMEDIA_TYPE_VIDEO) {
```

Comment:
Where `av_state_->codec_params_->codec_type`  from if `av_state_->codec_params_->codec_type` is false?